### PR TITLE
Feature/10266 fss UI implement sign in awareness for home search tab

### PR DESCRIPTION
**Pull request by Shrikan14443**
B2C login awareness for home page and search page.
Automated functional test.

### CODE REVIEW

**Review by bhalcha13813 — APPROVED** [review]
LGTM

File: Tests/FunctionalTests/fss-signin-page.spec.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+      popup.click(pageObjectsConfig.loginPopupNextButtonSelector)
+      await popup.waitForSelector(pageObjectsConfig.loginPopupSignInPasswordSelector)
+      popup.fill(pageObjectsConfig.loginPopupSignInPasswordSelector, password)
+      await popup.waitForSelector(pageObjectsConfig.loginPopupSignInButtonSelector)
+      popup.click(pageObjectsConfig.loginPopupSignInButtonSelector)     
+      
+    }  
+    //===============END===================================
+
+    it('User clicks Sign in link with valid credentails should display FirstName after login successfully', async () => {
```
Comment (gregzealley) [inline Tests/FunctionalTests/fss-signin-page.spec.ts:49]: First name or full name? Full name is used in the assertion.

Comment (gregzealley) [reply Tests/FunctionalTests/fss-signin-page.spec.ts:49]:
```suggestion
    it('User clicks Sign in link with valid credentials should display FirstName after login successfully', async () => {
```

File: Tests/FunctionalTests/fss-signin-page.spec.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            path: temppath + "\failedtest.jpeg"
+          });
+        }
+
+        await page.waitForSelector(pageObjectsConfig.loginAccountSelector);     
+        expect(await page.innerHTML(pageObjectsConfig.loginAccountLinkSelector)).toEqual(autoTestConfig.userFullName);    
+        
+      })
+
+      it('User clicks Sign in link with valid credentails should naviagte to search page after login successfully', async () => {
```
Comment (gregzealley) [inline Tests/FunctionalTests/fss-signin-page.spec.ts:68]:
```suggestion
      it('User clicks Sign in link with valid credentials should navigate to search page after login successfully', async () => {
```

File: Tests/FunctionalTests/fss-signin-page.spec.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    
+        page.click(pageObjectsConfig.loginSignInLinkSelector);
+        try {
+          await LoginPortal(autoTestConfig.user, autoTestConfig.password);
+        }
+        catch (e) {
+          let temppath = tmpdir();
+          
+          await page.screenshot({
+            path: temppath + "\failedtest.jpeg"
```
Comment (gregzealley) [inline Tests/FunctionalTests/fss-signin-page.spec.ts:59]: Would it be more useful to include the test name in the failed test JPG ?

File: Tests/FunctionalTests/fss-signin-page.spec.ts [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    
+        page.click(pageObjectsConfig.loginSignInLinkSelector);
+        await LoginPortal(autoTestConfig.user,autoTestConfig.password);  
+        
+        await page.waitForSelector(pageObjectsConfig.searchPageContainerHeaderSelector);    
+        expect(await page.innerHTML(pageObjectsConfig.searchPageContainerHeaderSelector)).toEqual(pageObjectsConfig.searchPageContainerHeaderText);    
+        
+      })
+
+      it('User clicks Search link with valid credentails should naviagte to search page after login successfully', async () => {
```
Comment (gregzealley) [inline Tests/FunctionalTests/fss-signin-page.spec.ts:78]:
```suggestion
      it('User clicks Search link with valid credentials should navigate to search page after login successfully', async () => {
```

File: Tests/FunctionalTests/fss-signin-page.spec.ts [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    
+        page.click(pageObjectsConfig.searchButtonSelector);
+        await LoginPortal(autoTestConfig.user,autoTestConfig.password);  
+        
+        await page.waitForSelector(pageObjectsConfig.searchPageContainerHeaderSelector);    
+        expect(await page.innerHTML(pageObjectsConfig.searchPageContainerHeaderSelector)).toEqual(pageObjectsConfig.searchPageContainerHeaderText);       
+        
+      })
+
+      it('When user navigate to search url without Sign in it should naviagte to fss home page', async () => {
```
Comment (gregzealley) [inline Tests/FunctionalTests/fss-signin-page.spec.ts:88]:
```suggestion
      it('When user navigate to search url without Sign in it should navigate to fss home page', async () => {
```

**Review by gregzealley — CHANGES_REQUESTED** [review]

File: Tests/FunctionalTests/fss-signin-page.spec.ts (hunk 5 above)
Comment (kajalkundu) [reply Tests/FunctionalTests/fss-signin-page.spec.ts]: Updated the typo error. Thanks.

File: Tests/FunctionalTests/fss-signin-page.spec.ts (hunk 4 above)
Comment (kajalkundu) [reply Tests/FunctionalTests/fss-signin-page.spec.ts]: Updated the typo error. Thanks.

File: Tests/FunctionalTests/fss-signin-page.spec.ts (hunk 2 above)
Comment (kajalkundu) [reply Tests/FunctionalTests/fss-signin-page.spec.ts]: Updated the typo error. Thanks.

File: Tests/FunctionalTests/fss-signin-page.spec.ts (hunk 1 above)
Comment (kajalkundu) [reply Tests/FunctionalTests/fss-signin-page.spec.ts]: Updated the typo error. Thanks.

File: Tests/FunctionalTests/fss-signin-page.spec.ts (hunk 3 above)
Comment (kajalkundu) [reply Tests/FunctionalTests/fss-signin-page.spec.ts]: Added Test name for the failed test. Thanks

**Review by perrytom — APPROVED** [review]
Looks good.

**Pull request merged**